### PR TITLE
image based lighting for PBR

### PR DESCRIPTION
**Pull request by sushantojal**

### CODE REVIEW

File: GVRf/Framework/framework/src/main/java/org/gearvrf/shaders/GVRPBRShader.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
     public GVRPBRShader(GVRContext gvrcontext)
     {
          super("float4 diffuse_color; float4 specular_color; float4 emissive_color; float metallic; float roughness; float specular_exponent; float lightmapStrength; float normalScale; float glossinessFactor",
-                "sampler2D diffuseTexture; sampler2D metallicRoughnessTexture; sampler2D specularTexture; sampler2D lightmapTexture; sampler2D diffuseTexture1; sampler2D normalTexture; sampler2D emissiveTexture",
+                "sampler2D diffuseTexture; sampler2D metallicRoughnessTexture; sampler2D specularTexture; sampler2D lightmapTexture; sampler2D diffuseTexture1; sampler2D normalTexture; sampler2D emissiveTexture; sampler2D brdfLUTTexture; samplerCube diffuseEnvTex; samplerCube specularEnvTexture",
```
Comment (apjagdale) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/shaders/GVRPBRShader.java]: Binding for texture descriptor starts from 5. Give binding locations for eg: diffuseTexture 5, metallicRoughnessTexture 6 and so on in shader.

File: GVRf/Framework/framework/src/main/res/raw/pbr_surface.fsh [hunk 2]
```diff
@@ -25,15 +25,41 @@ layout(set = 0, binding = 14) uniform sampler2D lightmapTexture;
 #endif
 
 #ifdef HAS_normalTexture
-layout(location = 9) in vec2 normal_coord;
-layout(set = 0, binding = 15) uniform sampler2D normalTexture;
+    layout(location = 9) in vec2 normal_coord;
```
Comment (apjagdale) [inline GVRf/Framework/framework/src/main/res/raw/pbr_surface.fsh]: location for normal_coord is 10

**Comment by apjagdale** [comment]
@sushantojal Could you also make binding = 5 for cubemap_frag.fsh for u_texture.

**Comment by liaxim** [comment]
Planning to merge on Wednesday.

Should https://github.com/gearvrf/GearVRf-Demos/pull/615 work with this pull request and Vulkan enabled?
```
06-05 08:49:04.416 28132 28176 I Adreno  : Program compilation failed
06-05 08:49:04.416 28132 28176 I Adreno  : Info log: Error: Bind group info mismatches the shader source for symbol u_texture
06-05 08:49:04.416 28132 28176 I Adreno  : 
```

**Comment by apjagdale** [comment]
@liaxim Yes the demo would work but you wont see the widget to switch to next model. I tested by modifying code to see each model.

**Comment by liaxim** [comment]
It crashes for me with the aforementioned shader compilation error.

**Comment by sushantojal** [comment]
Works if I rebase on top of the commit before 'Vulkan: Image Cleanup'. @apjagdale how should I change the bindings to be consistent with those in the image cleanup commit?

**Pull request merged**